### DISCUSSION TITLE
Binary encoding not working

### DISCUSSION THREAD

**Discussion by mellis481** [sindresorhus/got#1642] [Q&A] [answered]
I have the following code which I use to download an assortment of files from S3 and save them locally:

```
const responsePromise = got.get(urlToDownload);
const body = await responsePromise.text();
const fileStream = fs.createWriteStream(fullSavePath);
fileStream.write(body);
```

It works great, but when a `favicon.ico` file is downloaded, it's essentially corrupted because of bad encoding.  It would appear the source file is encoded with ANSI.  got's default is UTF-8.  The following change did nothing (the file still gets encoded in UTF-8):

```
const responsePromise = got.get(urlToDownload, { encoding: 'latin1' });
```

[This](https://github.com/sindresorhus/got/issues/268) isn't didn't offer any help.

**Comment by szmarczak** [comment, marked as answer]
https://github.com/sindresorhus/got#encoding

> Note: This doesn't affect streams! Instead, you need to do got.stream(...).setEncoding(encoding).

**Reply by szmarczak** [reply]
Try `encoding: 'binary'`

**Reply by szmarczak** [reply]
Also why are you doing `.text()` instead of `.buffer()`?
